includes/class-apple-news.php

```diff
@@ -161,32 +239,34 @@ public function migrate_api_settings( $wp_settings ) {
 	public function migrate_caption_settings( $wp_settings ) {
 
 		// Check for the presence of caption-specific settings.
-		if ( isset( $wp_settings['caption_color'] )
-		     && isset( $wp_settings['caption_font'] )
-		     && isset( $wp_settings['caption_line_height'] )
-		     && isset( $wp_settings['caption_size'] )
-		     && isset( $wp_settings['caption_tracking'] )
-		) {
+		if ( $this->_all_keys_exist( $wp_settings, array(
+			'caption_color',
+			'caption_font',
+			'caption_line_height',
+			'caption_size',
+			'caption_tracking',
+		) ) ) {
 			return $wp_settings;
 		}
 
-		// Migrate settings, as necessary.
-		$settings = array( 'color', 'font', 'line_height', 'size', 'tracking' );
-		foreach ( $settings as $setting ) {
-			$body_setting = 'body_' . $setting;
-			$caption_setting = 'caption_' . $setting;
-			if ( ! isset( $wp_settings[ $caption_setting ] )
-			     && isset( $wp_settings[ $body_setting ] )
-			) {
-				$wp_settings[ $caption_setting ] = $wp_settings[ $body_setting ];
-
-				// Adjust font size down by 2 to match legacy handling.
-				if ( 'size' === $setting ) {
-					$wp_settings[ $caption_setting ] -= 2;
-				}
-			}
+		// Clone and modify font size, if necessary.
+		if ( ! isset( $wp_settings['caption_size'] )
+		     && isset( $wp_settings['body_size'] )
+		) {
+			$wp_settings['caption_size'] = $wp_settings['body_size'] - 2;
```

Comment:
Not a bad idea to add a check ahead of time to be safe. Thanks for the suggestion.